test/TestProjects/MgmtReferenceTypes/Generated/Models/ErrorResponse.Serialization.cs

```diff
@@ -22,7 +22,7 @@ void IUtf8JsonSerializable.Write(Utf8JsonWriter writer)
             if (Optional.IsDefined(Error))
             {
                 writer.WritePropertyName("error"u8);
-                writer.WriteObjectValue(Error);
+                JsonSerializer.Serialize(writer, Error);
```

Comment:
The new code is somehow correct in a way but not entirely correct.

The type of `Error` property here is `Azure.ResponseError`. It does not implement the internal interface `IUtf8JsonSerializable` therefore the `writer.WriteObjectValue` extension method will just throw: https://github.com/Azure/autorest.csharp/blob/91b51d23bbd026b4169c1823080d0688818c192f/src/assets/Azure.Core.Shared/Utf8JsonWriterExtensions.cs#L73
In the meantime, the type `Azure.ResponseError` has a `JsonConverter` attribute on it, therefore `JsonSerializer.Serialize` is the correct method to call here.
On the other hand, why this is not entirely correct? Because the `JsonConverter` only implements the `Read` part, `Write` method of it just throws, therefore the invocation here also just throws.
It is not entirely correct, but it is closer.